blocks/editable/patterns.js

```diff
@@ -0,0 +1,241 @@
+			const regExp = new RegExp( '(.*)' + escapedDelimiter + '.+' + escapedDelimiter + '$' );
+			const match = textBeforeCaret.match( regExp );
+
+			if ( ! match ) {
+				return acc;
+			}
+
+			const startOffset = match[ 1 ].length;
+			const endOffset = carretOffset - item.delimiter.length;
+			const before = textBeforeCaret.charAt( startOffset - 1 );
+			const after = textBeforeCaret.charAt( startOffset + item.delimiter.length );
+			const delimiterFirstChar = item.delimiter.charAt( 0 );
+
+			// test*test* => format applied
+			// test *test* => applied
+			// test* test* => not applied
+			if ( startOffset && /\S/.test( before ) ) {
+				if ( /\s/.test( after ) || before === delimiterFirstChar ) {
+					return acc;
+				}
+			}
+
+			const contentRegEx = new RegExp( '^[\\s' + escapeRegExp( delimiterFirstChar ) + ']+$' );
+			const content = textBeforeCaret.slice( startOffset, endOffset );
+
+			// Do not replace when only whitespace and delimiter characters.
+			if ( contentRegEx.test( content ) ) {
+				return acc;
+			}
+
+			return {
+				start: startOffset,
+				pattern: item,
+			};
+		}, {} );
+
+		if ( ! pattern ) {
+			return;
+		}
+
+		const { delimiter, format } = pattern;
+		const formats = editor.formatter.get( format );
+
+		if ( ! formats || ! formats[ 0 ].inline ) {
+			return;
+		}
+
+		editor.undoManager.add();
+		editor.undoManager.transact( () => {
+			node.insertData( carretOffset, '\uFEFF' );
+
+			const newNode = node.splitText( start );
+			const zero = newNode.splitText( carretOffset - start );
+
+			newNode.deleteData( 0, delimiter.length );
+			newNode.deleteData( newNode.data.length - delimiter.length, delimiter.length );
+
+			editor.formatter.apply( format, {}, newNode );
+			editor.selection.setCursorLocation( zero, 1 );
+
+			// We need to wait for native events to be triggered.
+			setTimeout( () => {
+				canUndo = 'space';
+
+				editor.once( 'selectionchange', () => {
+					if ( zero ) {
+						const zeroOffset = zero.data.indexOf( '\uFEFF' );
+
+						if ( zeroOffset !== -1 ) {
+							zero.deleteData( zeroOffset, zeroOffset + 1 );
+						}
+					}
+				} );
+			} );
+		} );
+	}
+
+	function space() {
+		if ( ! onReplace ) {
+			return;
+		}
+
+		// Merge text nodes.
+		editor.getBody().normalize();
+
+		const content = getContent();
+
+		if ( ! content.length ) {
+			return;
+		}
+
+		const firstText = content[ 0 ];
+
+		const { result, pattern } = spacePatterns.reduce( ( acc, item ) => {
+			return acc.result ? acc : {
+				result: item.regExp.exec( firstText ),
+				pattern: item,
+			};
+		}, {} );
+
+		if ( ! result ) {
+			return;
+		}
+
+		const range = editor.selection.getRng();
+		const matchLength = result[ 0 ].length;
+		const remainingText = firstText.slice( matchLength );
+
+		// The caret position must be at the end of the match.
+		if ( range.startOffset !== matchLength ) {
+			return;
+		}
+
+		const block = pattern.transform( {
+			content: [ remainingText, ...drop( content ) ],
```

Comment:
I'm trying to track down what this `transform` function is meant to accept as an argument. Is the `content` object intended to be an array of two members, the first entry the text following the matching and the second entry the matched substring? Is it then expected that any pattern must match only at the start of a string? Where is this documented?